core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -437,7 +437,11 @@ class ListAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
       s"Cannot merge ${this.getClass.getName} with ${other.getClass.getName}")
   }
 
-  override def value: java.util.List[T] = java.util.Collections.unmodifiableList(_list)
+  override def value: java.util.List[T] = _list.synchronized {
+    val newList = new java.util.ArrayList[T]
```

Comment:
Why not go ahead and import the java classes? use a rename to disambiguate list. Or leave java.util.List alone but import the others?

Just `java.util.Collections.unmodifiableList(new java.util.ArrayList(_list))` should be better